dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/AzureOpenAIClientBase.cs

```diff
@@ -77,4 +77,22 @@ public abstract class AzureOpenAIClientBase : ClientBase
         this.ModelId = modelId;
         this.Client = new OpenAIClient(new Uri(endpoint), credential, options);
     }
+
+    /// <summary>
+    /// Creates a new AzureOpenAIClientBase client instance using the specified OpenAIClient
+    /// </summary>
+    /// <param name="modelId">Azure OpenAI model ID or deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="openAIClient">Custom <see cref="OpenAIClient"/>.</param>
+    /// <param name="logger">Application logger</param>
+    private protected AzureOpenAIClientBase(
+        string modelId,
+        OpenAIClient openAIClient,
```

Comment:
I see why you've done this, but it's breaking my brain.... We're passing in an OpenAIClient (from Azure SDK?) to an [AzureOpenAIClientBase](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/AzureOpenAIClientBase.cs), which is also a [ClientBase](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs), but not an [OpenAIClientBase](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/OpenAIClientBase.cs), or the other [OpenAIClientBase](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.AI.OpenAI/CustomClient/OpenAIClientBase.cs).

We really should take a closer look here and clean this up.

For this PR: Personally, I'd rather not expose the actual Azure SDK classes in our public API surface. If we were to pass through configuration or specific properties instead of OpenAIClient objects to meet the APIM needs, what would those be?

Reply:
@shawncal The `OpenAIClient` uses a class called `OpenAIClientOptions` so we would have to create out own abstraction for this which also feels wrong.
Why is not exposing Azure SDK classes in the public API important?

Reply:
Creating a new PR as this one has picked up some unexpected changes